pkg/services/ngalert/state/manager.go

```diff
@@ -99,11 +110,54 @@ func (st *Manager) Run(ctx context.Context) error {
 		case <-ctx.Done():
 			st.log.Debug("Stopping")
 			ticker.Stop()
+			if st.saveStateAsync {
```

Comment:
I really don't understand why we have a ticker for metric scraping loop here - I'll quickly file a PR to fix that.

Reply:
I have filed a PR here: https://github.com/grafana/grafana/pull/75296